index.js

```diff
@@ -1081,20 +1088,31 @@
           }
         }
 
-        // activates page tracking
-        if (trackRoutes) {
-          $rootScope.$on(pageEvent, function () {
-            // Apply $route based filtering if configured
-            if (readFromRoute) {
-              // Avoid tracking undefined routes, routes without template (e.g. redirect routes)
-              // and those explicitly marked as 'do not track'
-              if (!$route.current || !$route.current.templateUrl || $route.current.doNotTrack) {
+        // This section adds different listeners based on the configuration
+        var eventListener;
+        if (trackRouteFn != null) {
```

Comment:
Instead of checking for null, check that it is a function: `typeof trackRouteFn === 'function'`